TCutility/analysis/ts_vibration.py

```diff
@@ -0,0 +1,107 @@
+    basedist = atom_distance(base, a1+1, a2+1)
+    x = abs((atom_distance(pos, a1+1, a2+1)/basedist)-1)
+    y = abs((atom_distance(neg, a1+1, a2+1)/basedist)-1)
+    return (x+y)/2
+
+def determine_ts_reactioncoordinate(calc_dir: str, bond_tolerance: float = 1.28, min_delta_dist: float = 0.1) -> np.ndarray:
+    '''Function to retrieve reaction coordinate from a given transitionstate, using the first imaginary frequency.
+
+    Args:
+        calc_dir: path pointing to the desired calculation.
+        bond_tolerance: parameter for plams.molecule.guess_bonds() function
+        min_delta_dist: minimum relative bond length change before qualifying as active atom
+
+    Returns:
+        Array containing all the obtained reaction coordinates. Reaction coordinate format is [active_atom1, active_atom2, sign]
+        Symmetry elements are ignored, by convention the atom labels are increasing (atom1 < atom2)
+    '''
+
+    data = read(calc_dir)
+    assert 'modes' in data.properties.vibrations, 'Vibrational data is required, but was not present in .rkf file'
+
+    outputmol = data.molecule.output
+
+    base = np.array(outputmol).reshape(-1,3)
+    tsimode = np.array(data.properties.vibrations.modes[0]).reshape(-1,3)
+
+    posvib = outputmol.copy()
+    posvib.from_array(base + tsimode)
+    posvib.guess_bonds(dmax=bond_tolerance)
+    negvib = outputmol.copy()
+    negvib.from_array(base - tsimode)
+    negvib.guess_bonds(dmax=bond_tolerance)
+
+    pairs = np.where(posvib.bond_matrix() - negvib.bond_matrix())
+    rc = np.array(
+                [   [a+1, b+1, np.sign(posvib.bond_matrix()[a][b] - negvib.bond_matrix()[a][b])]
+                    for a, b in np.c_[pairs[0], pairs[1]] 
+                    if a < b and avg_relative_bond_length_delta(outputmol, posvib, negvib, a, b) > min_delta_dist
+                ],  dtype=int)
+    return rc
+
+
+def validate_transitionstate(calc_dir: str, rcatoms = None, bond_tolerance: float = 1.28, min_delta_dist: float = 0.1) -> bool:
+    '''Function to determine whether a transition state calculation yielded the expected transition state.
+
+    Args:
+        calc_dir: path pointing to the desired calculation.
+        rcatoms: list or array containing expected reaction coordinates, to check against the transition state. If not defined, it is obtained from the ams.rkf user input
+        bond_tolerance: parameter for plams.molecule.guess_bonds() function, used in determine_ts_reactioncoordinate
+        min_delta_dist: minimum relative bond length change before qualifying as active atom, used in determine_ts_reactioncoordinate
+
+    Returns:
+        Boolean value, True if the found transition state reaction coordinates match the expected reaction coordinates, False otherwise
+    '''
+
+    result = determine_ts_reactioncoordinate(calc_dir, bond_tolerance, min_delta_dist)
+
+    if isinstance(rcatoms, list):
```

Comment:
Use https://numpy.org/doc/stable/reference/generated/numpy.atleast_2d.html for this